internal/messaging/response_handler.go

```diff
@@ -1,147 +1,536 @@
 	if response.MintResponse.Header == nil {
 		response.MintResponse.Header = &typesv1alpha.ResponseHeader{}
 	}
 	if response.MintTransactionId == "" {
 		addErrorToResponseHeader(response, "missing mint transaction id")
 		return true
 	}
-	mintID, err := ids.FromString(response.MintTransactionId)
+
+	abi, err := loadABI(h.cfg.BookingTokenABIFile)
 	if err != nil {
-		addErrorToResponseHeader(response, fmt.Sprintf("error parsing mint transaction id: %v", err))
+		addErrorToResponseHeader(response, fmt.Sprintf("error loading ABI: %v", err))
 		return true
 	}
 
-	success, txID, err := h.tvmClient.SendTxAndWait(ctx, transferNFTAction(h.tvmClient.Address(), mintID))
+	value64 := uint64(response.BookingToken.TokenId)
+	tokenID := new(big.Int).SetUint64(value64)
+
+	bookingTokenAddress := common.HexToAddress("0xd4e2D76E656b5060F6f43317E8d89ea81eb5fF8D")
+
+	txID, err := buy(
+		h.ethClient,
+		bookingTokenAddress,
+		abi,
+		h.pk.ToECDSA(),
+		tokenID)
 	if err != nil {
 		errMessage := fmt.Sprintf("error buying NFT: %v", err)
 		if errors.Is(err, context.DeadlineExceeded) {
-			errMessage = fmt.Sprintf("%v: %v", tvm.ErrAwaitTxConfirmationTimeout, h.tvmClient.Timeout)
+			errMessage = fmt.Sprintf("%v", evm.ErrAwaitTxConfirmationTimeout)
 		}
 		addErrorToResponseHeader(response, errMessage)
 		return true
 	}
-	if !success {
-		addErrorToResponseHeader(response, "buying NFT failed")
-		return true
-	}
 
-	h.logger.Infof("Bought NFT (txID=%s) with ID: %s\n", txID, mintID)
-	response.BuyTransactionId = txID.String()
+	h.logger.Infof("Bought NFT (txID=%s) with ID: %s\n", txID, response.MintTransactionId)
+	response.BuyTransactionId = txID
 	return false
 }
 
-func addErrorToResponseHeader(response *ResponseContent, errMessage string) {
-	response.MintResponse.Header.Status = typesv1alpha.StatusType_STATUS_TYPE_FAILURE
-	response.MintResponse.Header.Alerts = append(response.MintResponse.Header.Alerts, &typesv1alpha.Alert{
-		Message: errMessage,
-		Type:    typesv1alpha.AlertType_ALERT_TYPE_ERROR,
-	})
+// Loads an ABI file
+func loadABI(filePath string) (abi.ABI, error) {
+	file, err := os.ReadFile(filePath)
+	if err != nil {
+		return abi.ABI{}, err
+	}
+	return abi.JSON(strings.NewReader(string(file)))
+}
+
+// Registers a new supplier with the BookingToken contract
+func register(client *ethclient.Client, contractABI abi.ABI, privateKey *ecdsa.PrivateKey, supplierName string, bookingTokenAddress common.Address) error {
+	address := crypto.PubkeyToAddress(privateKey.PublicKey)
+	nonce, err := client.PendingNonceAt(context.Background(), address)
+	if err != nil {
+		return err
+	}
+
+	gasPrice, err := client.SuggestGasPrice(context.Background())
+	if err != nil {
+		return err
+	}
+
+	// Check supplier name and set default if empty
+	if supplierName == "" {
+		defaultSupplierName := "Default Supplier"
+		fmt.Printf("Supplier name cannot be empty. Setting name to: %s \n", defaultSupplierName)
```

Comment:
I'd suggest to replace all fmt lines with log lines, since fmt is there only cause example didn't have any existing logger
See similar places